sphinx/texinputs/sphinx.sty

```diff
@@ -65,6 +65,8 @@
 \DeclareStringOption[-1]{numfigreset}
 \DeclareBoolOption[false]{nonumfigreset}
 \DeclareBoolOption[false]{mathnumfig}
+\define@key{sphinx}{bookmarksdepth}{\AtBeginDocument{\hypersetup{bookmarksdepth=#1}}}
+\AtBeginDocument{\define@key{sphinx}{bookmarksdepth}{\hypersetup{bookmarksdepth=#1}}}
```

Comment:
we can not use ``\hypersetup`` before hyperref is loaded and we can not use ``\PassOptionsToPackage`` after hyperref is loaded. The initial definition could do some `\@ifpackageloaded{hyperref}` but it looked complicated for nothing. Manual `\hypersetup{bookmarksdepth=..}` in preamble will thus be overridden by a usage of bookmarksdepth sphinxsetup key, but user should use the sphinxsetup key.

Reply:
> we can not use `\hypersetup` before hyperref is loaded and we can not use `\PassOptionsToPackage` after hyperref is loaded. 

(background here is that the key allows usage of `\sphinxsetup`  macro, so my comment refers to using directly `\sphinxsetup` macro in preamble, rather than indirectly via conf.py's ``latex_elements['sphinxsetup']``)